core/src/test/scala/org/locationtech/rasterframes/RasterFramesStatsSpec.scala

```diff
@@ -21,46 +21,109 @@
 
 package org.locationtech.rasterframes
 
-import org.apache.spark.sql.functions.col
-
-import org.locationtech.rasterframes._
-import org.locationtech.rasterframes.RasterFunctions
+import org.apache.spark.sql.functions.{col, explode}
 
 class RasterFramesStatsSpec extends TestEnvironment with TestData {
 
-  describe("DataFrame.tileStats extension methods") {
+  import spark.implicits._
 
-    val df = TestData.sampleGeoTiff.toDF()
-        .withColumn("tilePlus2", rf_local_add(col("tile"), 2))
+  val df = TestData.sampleGeoTiff
+    .toDF()
+    .withColumn("tilePlus2", rf_local_add(col("tile"), 2))
 
-    it("should compute approx percentiles for a single tile col"){
+  describe("DataFrame.tileStats extension methods") {
+    it("should compute approx percentiles for a single tile col") {
 
-      val result = df.tileStat().approxTileQuantile(
-        "tile",
-        Array(0.10, 0.50, 0.90),
-        0.00001
-      )
+      val result = df
+        .tileStat()
+        .approxTileQuantile(
+          "tile",
+          Array(0.10, 0.50, 0.90),
+          0.00001
+        )
 
-      result.length should be (3)
+      result.length should be(3)
 
       // computing externally with numpy we arrive at 7963, 10068, 12160 for these quantiles
-      result should contain inOrderOnly (7963.0, 10068.0, 12160.0)
+      result should contain inOrderOnly(7963.0, 10068.0, 12160.0)
     }
 
-    it("should compute approx percentiles for many tile cols"){
-      val result = df.tileStat().approxTileQuantile(
-        Array("tile", "tilePlus2"),
-        Array(0.25, 0.75),
-        0.00001
-      )
-      result.length should be (2)
+    it("should compute approx percentiles for many tile cols") {
```

Comment:
Note: I didn't replicate the "many columns" tests because I have learned after several function implementations is that you don't want to operate over multiple columns unless you can help it (i.e `Generator`s) because the query plan optimizer can't cull the column if it's not used downstream. We might be able to simulate the approach by providing a convenience method that performs the `map` over input columns for the user.

Reply:
not to be worried about in this context. the many columns operation makes sense for `df.stats.some_stat(col1, col2, col3)` kind of API